quickwit-proto/proto/metastore_api.proto

```diff
@@ -23,44 +23,62 @@ syntax = "proto3";
 package quickwit_metastore_api;
 
 service MetastoreApiService {
-  // Create an index.
+  // Creates an index.
   rpc create_index(CreateIndexRequest) returns (CreateIndexResponse);
 
-  // Get an index metadata.
+  // Gets an index metadata.
   rpc index_metadata(IndexMetadataRequest) returns (IndexMetadataResponse);
 
-  // Get an indexes metadatas.
+  // Gets an indexes metadatas.
   rpc list_indexes_metadatas(ListIndexesMetadatasRequest) returns (ListIndexesMetadatasResponse);
 
-  // Delete an index
+  // Deletes an index
   rpc delete_index(DeleteIndexRequest) returns (DeleteIndexResponse);
 
-  // Get all splits from index.
+  // Gets all splits from index.
   rpc list_all_splits(ListAllSplitsRequest) returns (ListSplitsResponse);
 
-  // Get splits from index.
+  // Gets splits from index.
   rpc list_splits(ListSplitsRequest) returns (ListSplitsResponse);
 
-  // Stage split.
+  // Stages split.
   rpc stage_split(StageSplitRequest) returns (SplitResponse);
 
-  // Publish split.
+  // Publishes split.
   rpc publish_splits(PublishSplitsRequest) returns (SplitResponse);
 
-  // Mark splits for deletion.
+  // Marks splits for deletion.
   rpc mark_splits_for_deletion(MarkSplitsForDeletionRequest) returns (SplitResponse);
 
-  // Delete splits.
+  // Deletes splits.
   rpc delete_splits(DeleteSplitsRequest) returns (SplitResponse);
 
-  // Add source.
+  // Adds source.
   rpc add_source(AddSourceRequest) returns (SourceResponse);
 
-  // Remove source.
+  // Removes source.
   rpc delete_source(DeleteSourceRequest) returns (SourceResponse);
 
   // Resets source checkpoint.
   rpc reset_source_checkpoint(ResetSourceCheckpointRequest) returns (SourceResponse);
+
+  // Gets last opstamps by index id.
+  rpc last_delete_opstamps_by_index_id(LastDeleteOpstampsByIndexIdRequest) returns (LastDeleteOpstampsByIndexIdResponse);
+
+  // Creates a delete task.
+  rpc create_delete_task(DeleteQueryRequest) returns (DeleteTaskResponse);
+
+  // Deletes all delete tasks for a given `index_id`.
+  rpc delete_delete_tasks(DeleteDeleteTasksRequest) returns (DeleteTasksResponse);
+
+  // Updates splits `delete_opstamp`.
+  rpc update_splits_delete_opstamp(UpdateSplitsDeleteOpstampRequest) returns (UpdateSplitsDeleteOpstampResponse);
```

Comment:
can you confirm that this is used when there are no deleted docs, and that the other  path works using deletes.

Reply:
Do you mean confirm by some tests?

Reply:
ah no I just meant explaining me... but I think I saw that in the code.